agent/api/container/container.go

```diff
@@ -109,15 +106,25 @@ type HealthStatus struct {
 	Output string `json:"output,omitempty"`
 }
 
-type ManagedAgent struct {
+type ManagedAgentState struct {
+	// ID of this managed agent state
+	ID string `json:"id:omitempty"`
 	// Status is the managed agent health status
 	Status apicontainerstatus.ManagedAgentStatus `json:"status:omitempty"`
-	// Name is always executeCommandAgent (until a new agent type is introduced)
-	Name string `json:"name,omitempty"`
 	// Reason is a placeholder for failure messaging
 	Reason string `json:"reason,omitempty"`
 	// LastStartedAt is the timestamp when the status last went from PENDING->RUNNING
 	LastStartedAt time.Time `json:"lastStartedAt,omitempty"`
+	// Metadata holds metadata about the managed agent
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
```

Comment:
feels like it would be a lot easier to read if metadata was a struct and to avoid the extra toMap conversion

Reply:
The problem is that in the future, different managed agents will have different metadata requirements, defining a struct in the container would limit to the metadata of exec cmd only, thus making it generic through a map.

Reply:
ah ok, that makes sense